src/controller/proxy/remote.go

```diff
@@ -0,0 +1,84 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package proxy
+
+import (
+	"github.com/docker/distribution"
+	"github.com/goharbor/harbor/src/replication/adapter"
+	"github.com/goharbor/harbor/src/replication/registry"
+	"io"
+)
+
+// RemoteInterface defines operations related to remote repository under proxy
+type RemoteInterface interface {
```

Comment:
Does it need to be exposed?

Reply:
Looks like this interface can be replaced by a func to fetch registry adapter by ID.